tests/backup/backup_scale_test.go

```diff
@@ -134,24 +134,17 @@ var _ = Describe("{MultipleBackupLocationWithSameEndpoint}", func() {
 			log.InfoD("Taking restore for each backups created from px-admin")
 			ctx, err := backup.GetAdminCtxFromSecret()
 			log.FailOnError(err, "failed to fetch ctx for admin")
-			createRestore := func(backupName string, restoreName string, namespace string) {
-				defer GinkgoRecover()
-				defer wg.Done()
-				customNamespace := "custom-" + namespace + RandomString(4)
-				namespaceMapping := map[string]string{namespace: customNamespace}
-				appContextsToBackup := FilterAppContextsByNamespace(scheduledAppContexts, []string{namespace})
-				err = CreateRestoreWithValidation(ctx, restoreName, backupName, namespaceMapping, make(map[string]string), DestinationClusterName, BackupOrgID, appContextsToBackup)
-				dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying creation of restore %s of backup %s", restoreName, backupName))
-				restoreNames = SafeAppend(&mutex, restoreNames, restoreName).([]string)
-			}
 			for index := 0; index < numberOfBackups; index++ {
 				for backupName, namespace := range userBackupMap[index] {
-					wg.Add(1)
 					restoreName := fmt.Sprintf("%s-%s", RestoreNamePrefix, backupName)
-					go createRestore(backupName, restoreName, namespace)
+					customNamespace := "custom-" + namespace + RandomString(5)
+					namespaceMapping := map[string]string{namespace: customNamespace}
+					appContextsToBackup := FilterAppContextsByNamespace(scheduledAppContexts, []string{namespace})
+					err = CreateRestoreWithValidation(ctx, restoreName, backupName, namespaceMapping, make(map[string]string), DestinationClusterName, BackupOrgID, appContextsToBackup)
+					dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying creation of restore %s of backup %s", restoreName, backupName))
+					restoreNames = SafeAppend(&mutex, restoreNames, restoreName).([]string)
```

Comment:
Can we do something like what is done in testcase :MultipleCustomRestoreSameTimeDiffStorageClassMapping
Take parallel restores and the validate is seperately? 
Or will it lead to testcase failure?

Reply:
@snigdha-px , Just checked that code, The issue here was the same context switch during validation.
Ill implement the changes to do it separately.

Reply:
Thanks for suggestion @snigdha-px .
Testing log after changes.
https://jenkins.pwx.dev.purestorage.com/job/portworx-backup/job/system-tests/job/byoc/job/px-backup-on-demand-system-test-byoc/4734/console